### PR TITLE
Use temporal-activities archive extractor

### PR DESCRIPTION
**Pull request by djjuhasz**
Use github.com/artefactual-sdps/temporal-activities/archive to extract archive contents instead of custom unarchive activity.

### CODE REVIEW

File: internal/config/config.go [hunk 1]
```diff
@@ -35,6 +36,7 @@ type Configuration struct {
 	A3m           a3m.Config
 	AM            am.Config
 	API           api.Config
+	Archive       archive.Config
```
Comment (jraddaoui) [inline internal/config/config.go]: Can we add a configuration example (even if it is with the defaults) to `enduro.toml`? I think a section called `[archive]` may not be clear enough, so we may want to change this field to `ExtractArchiveActivity` or something like that.

File: internal/workflow/processing.go [hunk 2]
```diff
@@ -328,22 +335,26 @@ func (w *ProcessingWorkflow) SessionHandler(
...
-				StripTopLevelDir: tinfo.req.StripTopLevelDir,
-			},
+			archive.ExtractActivityName,
+			&archive.ExtractActivityParams{SourcePath: tinfo.TempPath},
 		).Get(activityOpts, &result)
 		if err != nil {
-			return err
+			switch err {
+			case archive.ErrInvalidArchive:
+				break // Not an archive file, bundle the source file as-is.
```
Comment (jraddaoui) [inline internal/workflow/processing.go]: The `break` is implied in Go.

File: internal/workflow/processing.go [hunk 3]
```diff
@@ -328,22 +335,26 @@ func (w *ProcessingWorkflow) SessionHandler(
...
+			default:
+				return temporal_tools.NewNonRetryableError(err)
+			}
+		}
+		if err == nil {
```
Comment (jraddaoui) [inline internal/workflow/processing.go]: `else`.

**Review by jraddaoui — CHANGES_REQUESTED** [review]
Looking great @djjuhasz! Just a minor concern about the configuration.

**Review by jraddaoui — APPROVED** [review]
Thanks @djjuhasz!

**Comment by djjuhasz** [comment]
> Thanks @djjuhasz!

Oops, didn't see your approval. Thanks @jraddaoui! :grin:

**Pull request merged**